gems/sorbet-runtime/lib/types/private/methods/signature.rb

```diff
@@ -67,14 +65,17 @@ def initialize(method:, method_name:, raw_arg_types:, raw_return_type:, bind:, m
     writer_method = declared_param_names != SIG_EMPTY_DECLARED_PARAMETERS && parameters == UNNAMED_REQUIRED_PARAMETERS && method_name[-1] == "="
     # For writer methods, map the single parameter to the method name without the "=" at the end
     parameters = [[:req, method_name[0...-1].to_sym]] if writer_method
-    param_names = parameters.map {|_, name| name}
-    missing_names = param_names - declared_param_names
-    if !missing_names.empty?
+    is_name_missing = parameters.any? {|_, name| !raw_arg_types.key?(name)}
```

Comment:
I really like this bit of the change.  But can we go further and just get rid of `declared_param_names` entirely until we need it?  (I think some of the error cases below need it, but otherwise, we can get by with stuff like this.)